### PR TITLE
Timestamp Ordering

### PR DESCRIPTION
**Pull request by l-monninger**
# Summary
- **RFCs**: $\emptyset$.
- **Categories**: `protocol-units`

Corrects timestamp ordering and adds test. 

# Changelog

# Testing

# Outstanding issues
1. I feel like this was also corrected and tested in another PR. I'm going to check for that to link and make sure we aren't doing some unhigenic things here

### CODE REVIEW

File: protocol-units/mempool/util/src/lib.rs [hunk 1]
```diff
@@ -147,12 +147,12 @@ impl PartialOrd for MempoolTransaction {
 impl Ord for MempoolTransaction {
 	fn cmp(&self, other: &Self) -> Ordering {
 		// First, compare by slot_seconds
-		match self.slot_seconds.cmp(&other.slot_seconds) {
```
Comment (mzabaluev) [inline protocol-units/mempool/util/src/lib.rs:150]: Good!
Is the `slot_seconds` field left with any use whatsoever now?

File: protocol-units/settlement/mcr/client/Cargo.toml [hunk 2]
```diff
@@ -46,8 +50,6 @@ serde = { workspace = true }
 godfig = { workspace = true }
 
 dot-movement = { workspace = true }
-
-[dev-dependencies]
```
Comment (mzabaluev) [inline protocol-units/settlement/mcr/client/Cargo.toml:50]: Why remove this? We don't want test dependencies to pollute the build graph for anything using the mcr client crate.

File: process-compose/mcr-client/process-compose.test.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ processes:
   mcr-client-tests:
     command: |
       set -e
-      cargo test -p mcr-settlement-client --features="e2e" -- --nocapture
+      cargo run --bin test-mcr-settlement-client
```
Comment (mzabaluev) [inline process-compose/mcr-client/process-compose.test.yml:10]: This commit is unrelated to the timestamps. Was it added to fix the CI for this PR?

**Review by mzabaluev — CHANGES_REQUESTED** [review]

File: protocol-units/mempool/util/src/lib.rs (hunk 1 above)
Comment (l-monninger) [reply protocol-units/mempool/util/src/lib.rs:150]: Only insofar as we might want to make these structs self-describing.

File: process-compose/mcr-client/process-compose.test.yml (hunk 3 above)
Comment (l-monninger) [reply process-compose/mcr-client/process-compose.test.yml:10]: Yeah, CI fix.

File: protocol-units/settlement/mcr/client/Cargo.toml (hunk 2 above)
Comment (mzabaluev) [reply protocol-units/settlement/mcr/client/Cargo.toml:50]: Ah yes, that's why I don't like tests as binaries. Consider moving them to a separate crate e.g. mcr-settlement-tests, or finding a way to have them as integration tests in the crate `./tests` directory.

File: protocol-units/mempool/util/src/lib.rs [hunk 4]
```diff
@@ -147,12 +147,12 @@ impl PartialOrd for MempoolTransaction {
 impl Ord for MempoolTransaction {
 	fn cmp(&self, other: &Self) -> Ordering {
 		// First, compare by slot_seconds
```
Comment (mzabaluev) [inline protocol-units/mempool/util/src/lib.rs:149]: This comment is outdated now.
```suggestion
		// First, compare by timestamps
```

File: protocol-units/mempool/util/src/lib.rs (hunk 1 above)
Comment (mzabaluev) [reply protocol-units/mempool/util/src/lib.rs:150]: OK, I missed `Debug`. Is it just for that?

File: protocol-units/settlement/mcr/client/Cargo.toml (hunk 2 above)
Comment (mzabaluev) [reply protocol-units/settlement/mcr/client/Cargo.toml:50]: Or, perhaps, do add an "integration-tests" feature flag guarding these binaries, like we have it elsewhere, and make these dependencies conditional on that.

File: protocol-units/mempool/util/src/lib.rs (hunk 1 above)
Comment (l-monninger) [reply protocol-units/mempool/util/src/lib.rs:150]: I wasn't thinking about it just in terms of debug.

File: protocol-units/settlement/mcr/client/Cargo.toml (hunk 2 above)
Comment (0xmovses) [reply protocol-units/settlement/mcr/client/Cargo.toml:50]: I think we should be able to exclude the testing bin from the release build if we, in the root Cargo.toml:
```rust
[profile.release]
exclude = ["test-mcr-settlement-client"]
```

**Review by 0xmovses — CHANGES_REQUESTED** [review]
Logic looks good, one suggestion re. slim release binaries.

File: protocol-units/settlement/mcr/client/Cargo.toml (hunk 2 above)
Comment (l-monninger) [reply protocol-units/settlement/mcr/client/Cargo.toml:50]: I'm not familiar with the above syntax. I couldn't find it in the docs either. 

I would assume this also wouldn't change the bin size if it does work, just drop that particular binary from the workspace build. 

I couldn't actually find a good way to do this with profiles, so our best option is indeed to stick with a flag and just make sure everyone is using it. Sound good to you?

Comment (0xmovses) [reply protocol-units/settlement/mcr/client/Cargo.toml:50]: Ah, I meant it would not include the binary in the release build. But fine with the flag.

Comment (mzabaluev) [reply protocol-units/settlement/mcr/client/Cargo.toml:50]: I'm not sure about profiles, it's a weird way to affect the build graph.

I agree that the feature flag is not the best solution, either.
So my alternative suggestions are:
1) a separate test-only crate, or
2) find a way to build and run it as regular Rust tests.

It's less important that the tests require external setup.

Comment (l-monninger) [reply protocol-units/settlement/mcr/client/Cargo.toml:50]: I think we can move this to a separate PR: https://github.com/movementlabsxyz/movement/discussions/643

**Pull request merged**